README.md

```diff
@@ -1,16 +1,23 @@
-# Kubernetes Template Project
+# csi lib-iscsi
 
-The Kubernetes Template Project is a template for starting new projects in the GitHub organizations owned by Kubernetes. All Kubernetes projects, at minimum, must have the following files:
+A simple go package intended to assist CSI plugin authors by providing a tool set to manage iscsi connections.
 
-- a `README.md` outlining the project goals, sponsoring sig, and community contact information
-- an `OWNERS` with the project leads listed as approvers ([docs on `OWNERS` files][owners])
-- a `CONTRIBUTING.md` outlining how to contribute to the project
-- an unmodified copy of `code-of-conduct.md` from this repo, which outlines community behavior and the consequences of breaking the code
-- a `LICENSE` which must be Apache 2.0 for code projects, or [Creative Commons 4.0] for documentation repositories, without any custom content
-- a `SECURITY_CONTACTS` with the contact points for the Product Security Team 
-  to reach out to for triaging and handling of incoming issues. They must agree to abide by the
-  [Embargo Policy](https://github.com/kubernetes/sig-release/blob/master/security-release-process-documentation/security-release-process.md#embargo-policy) 
-  and will be removed and replaced if they violate that agreement.
+## Goals
+
+Provide a basic, lightweight library for CSI Plugin Authors to leverage some of the common tasks like connecting
+and disconnecting iscsi devices to a node.
+
+We inentionally avoid pulling in additional dependencies, and we intend to be stateless
+and as such are not using receivers.  Currently the focus is strictly based on a CSI context.
+
+## Design Philosophy
+
+The idea is to keep this as lightweight and generic as possible.  We intentionally avoid the use of any third party
+libraries or packages in this project.  We don't have a vendor directory, because we attempt to rely only on the std
+golang libs.  This may prove to not be ideal, and may be changed over time, but initially it's a worthwhile goal.
+
+Note that we're not using glog, or logrus, that's sure to raise hackles for folks.  We probably don't need them, and
+we may or may not even need logging in the lib at all.  For those that require logrus or glog however the simple
 
 ## Community, discussion, contribution, and support
 
```

Comment:
I'm not sure what the standard is for examples.  Do they go in the readme, or is the GoDoc enough?

Reply:
Yeah, I thought the "example" dir might be nice, but maybe instead just put in the README?  Probably makes more sense.  I'll give it another day for people to chime in, then I'll update as per your comments and merge this.